packages/baggage-span-processor/src/baggage-span-processor.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Context, propagation } from '@opentelemetry/api';
+import {
+  NoopSpanProcessor,
+  Span,
+} from '@opentelemetry/sdk-trace-base';
+
+/**
+ * The BaggageSpanProcessor reads entries stored in {@link Baggage}
+ * from the parent context and adds the baggage entries' keys and
+ * values to the span as attributes on span start.
+ *
+ * Add this span processor to a tracer provider.
+ *
+ * Keys and values added to Baggage will appear on subsequent child
+ * spans for a trace within this service *and* be propagated to external
+ * services in accordance with any configured propagation formats
+ * configured. If the external services also have a Baggage span
+ * processor, the keys and values will appear in those child spans as
+ * well.
+ *
+ * ⚠ Warning ⚠️
+ *
+ * Do not put sensitive information in Baggage.
+ *
+ * To repeat: a consequence of adding data to Baggage is that the keys and
+ * values will appear in all outgoing HTTP headers from the application.
+ */
+export class BaggageSpanProcessor extends NoopSpanProcessor {
```

Comment:
Why is extending NoopSpanProcessor, feels a little weird is extending something that should do nothing? maybe using SpanProcessor instead?

Reply:
I went with it so I didn't need to implement no-op's for flush, onend and shutdown. I agree it's a little weird though, I'll update to use the `SpanProcessor` interface instead.